module/Finna/src/Finna/RecordDriver/SolrEad3.php

```diff
@@ -392,6 +394,39 @@ public function getOriginationExtended(): array
         return $localeResults ?: $results;
     }
 
+    /**
+     * See if holdings tab is shown on current item
+     *
+     * @return bool
+     */
+    public function archiveRequestAllowed()
+    {
+        $xml = $this->getXmlRecord();
+        if ($xml) {
+            $attributes = $xml->attributes();
+            if (isset($attributes->level)) {
+                $datasourceSettings = $this->datasourceSettings[$this->getDataSource()];
+                if ($datasourceSettings) {
+                    // Requests only allowed on specified datasource
+                    if ($datasourceSettings['allowArchiveRequest']) {
+                        // Check if required filing unit exists and specified item hierarchy levels match the item's
+                        if ($datasourceSettings['filinUnitRequired'] && empty($this->getFilingUnit())) {
+                            return false;
+                        } elseif ($datasourceSettings['requestFormRecordLevels']) {
+                            $recordLevels = explode(';', $datasourceSettings['requestFormRecordLevels']);
+                            if (
+                                in_array($attributes->level, $recordLevels)
+                            ) {
+                                return true;
```

Comment:
Tässä pitäisi olla niin päin, että jos level ei ole oikea, palautetaan false. Sitten kaikkien ehtojen tarkistuksen jälkeen palautetaan true.